outbox.js

```diff
@@ -0,0 +1,205 @@
+class DistributedOutbox extends HTMLElement {
+  constructor() {
+    super();
+    this.renderedItems = new Set(); // Tracks rendered items
+    this.numPosts = 1; // Default value
+    this.page = 1; // Default value
+    this.totalPages = 0; // Keep track of total pages
+  }
+
+  static get observedAttributes() {
+    return ["url", "num-posts", "page"];
+  }
+
+  connectedCallback() {
+    // Use attributes or default values
+    this.numPosts = parseInt(this.getAttribute("num-posts")) || this.numPosts;
+    this.page = parseInt(this.getAttribute("page")) || this.page;
+    this.loadOutbox(this.getAttribute("url"));
+  }
+
+  async loadOutbox(outboxUrl) {
+    this.clearContent();
+    for await (const item of this.fetchOutboxItems(outboxUrl)) {
+      // Check if the item hasn't been rendered
+      if (!this.renderedItems.has(item.object)) {
```

Comment:
Is this supposed to remove duplicate posts? Two objects with the same data but loaded from different JSON.parse calls won't deduplicate. Instead you might want to use a Map which keys the object.id. This wouldn't preserve order. It might be good to get rid of this if and use an array and be okay when there's duplicates.

Reply:
![Screenshot 2024-02-22 at 2 04 04 AM](https://github.com/hyphacoop/reader.distributed.press/assets/68826419/0c9ceaf3-a63c-44e9-b4e7-b987d407c3ee)